frontend/simple/main.js

```diff
@@ -17,7 +18,39 @@ Vue.use(Router)
 Vue.use(VeeValidate)
 
 Vue.events = new Vue() // global event bus, use: https://vuejs.org/v2/api/#Instance-Methods-Events
-
+/*
+  The following are reusable guard for routes
+  the 'guard' defines how the route is blocked and the redirect determines the redirect behavior
+  when a route is blocked
+ */
+// Check if user is logged in
+var loginGuard = {
+  guard (store) {
+    return !store.state.loggedIn
+  },
+  redirect (to, from) {
+    return { path: '/signup', query: { next: to.path } }
+  }
+}
+// Check if user has a group to invite users to
+var inviteGuard = {
+  guard (store) {
+    return !store.state.currentGroupId
+  },
+  redirect (to, from) {
+    return { path: '/new-group' }
+  }
+}
```

Comment:
There are two things that are among our biggest enemies: line count (LOC), and complexity.

Excess LOC, just visually gives (at least my brain) the impression of excess complexity.

A mind trick I've picked up is that if something is in fact really simple, you can enforce and emphasize simplicity from a visual perspective by reducing the LOC.

One example of this is: if it's possible to define a function on a single line without hitting the ~81 character width limit, then do so. It makes it easier to mentally "chunk" and parse the code, as well as the rest of the code in the file.

So that is in fact the case with these entry guards. So this section could be collapsed a bit, to reduce LOC, while still being readable, by writing it this way:

```js
var loginGuard = {
  guard (store) { return !store.state.loggedIn },
  redirect (to, from) { return { path: '/signup', query: { next: to.path } } }
}
// Check if user has a group to invite users to
var inviteGuard = {
  guard (store) { return !store.state.currentGroupId },
  redirect (to, from) { return { path: '/new-group' } }
}
```